examples/holt_winters.py

```diff
@@ -0,0 +1,198 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+r"""
+Example: Holt-Winters Exponential Smoothing
+===========================================
+
+In this example we show how to implement Exponential Smoothing.
+This is intended to be a simple counter-part to the
+`Time Series Forecasting <https://num.pyro.ai/en/stable/tutorials/time_series_forecasting.html>`_
+notebook.
+
+The idea is that we have some times series
+
+.. math::
+
+    y_1, ..., y_T, y_{T+1}, ..., y_{T+H}
+
+where we train on :math:`y_1, ..., y_T` and predict for :math:`y_{T+1}, ..., y_{T+H}`,
+where :math:`T` is the maximum training timestamp and :math:`H` is the maximum number of
+future timesteps for which we want to forecast.
+
+We will be using the update equations from the excellent book
+`Forecasting Principles and Practice <https://otexts.com/fpp3/holt-winters.html>`_:
+
+.. math::
+
+    \hat{y}_{t+h|t} = l_t + hb_t + s_{t+h-m(k+1)}
+
+    l_t = \alpha(y_t - s_{t-m}) + (1-\alpha)(l_{t-1} + b_{t-1})
+
+    b_t = \beta^*(l_t-l_{t-1}) + (1-\beta^*)b_{t-1}
+
+    s_t = \gamma(y_t-l_{t-1}-b_{t-1})+(1-\gamma)s_{t-m}
+
+where
+
+* :math:`\hat{y}_t` is the forecast at time :math:`t`;
+* :math:`h` is the number of time steps into the future which we want to predict for;
+* :math:`l_t` is the level, :math:`b_t` is the trend,
+  and :math:`s_t` is the seasonality,
+* :math:`\alpha` is the level smoothing, :math:`\beta^*` is the trend
+  smoothing, and :math:`\gamma` is the seasonality smoothing.
+* :math:`k` is the integer part of :math:`(h-1)/m` (this looks more complicated than it is,
+  it just takes the latest seasonality estimate for this time point).
+
+.. image:: ../_static/img/examples/holt_winters.png
+    :align: center
+"""
+
+import argparse
+import os
+import time
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+
+import jax
+from jax import random
+import jax.numpy as jnp
+
+import numpyro
+from numpyro.contrib.control_flow import scan
+from numpyro.diagnostics import hpdi
+import numpyro.distributions as dist
+from numpyro.infer import MCMC, NUTS, Predictive
+
+matplotlib.use("Agg")
+
+N_POINTS_PER_UNIT = 10  # number of points to plot for each unit interval
+
+
+def holt_winters(y, n_seasons, future=0):
+    T = y.shape[0]
+    level_smoothing = numpyro.sample("level_smoothing", dist.Beta(1, 1))
+    trend_smoothing = numpyro.sample("trend_smoothing", dist.Beta(1, 1))
+    seasonality_smoothing = numpyro.sample(
+        "seasonality_smoothing", dist.Uniform(0, 1 - level_smoothing)
```

Comment:
It is better to reparameterize this to have static supports. You can either do it manually or through a transformed distributions:
```python
    with numpyro.handlers.reparam(config=lambda _: numpyro.infer.reparam.TransformReparam()):
        seasonality_smoothing = numpyro.sample("seasonality_smoothing", dist.TransformedDistribution(
            dist.Uniform(0, 1), dist.transforms.AffineTransform(0, 1 - level_smoothing))
        "seasonality_smoothing", dist.Uniform(0, 1 - level_smoothing)
```

Reply:
OK, I've gone with the other formulation from [fpp3](https://otexts.com/fpp3/holt-winters.html), which has `gamma_star` between 0 and 1, and `gamma` is `gamma_star * (1-alpha)`, is this OK?

Reply:
Thanks, LGTM!